packages/vector_graphics_compiler/lib/src/paint.dart

```diff
@@ -140,6 +140,18 @@ class LinearGradient extends Shader {
 
   @override
   LinearGradient applyBounds(Rect bounds, AffineMatrix transform) {
+    if (unitMode == GradientUnitMode.userSpaceOnUse) {
+      return LinearGradient(
+        from: transform.transformPoint(from),
+        to: transform.transformPoint(to),
+        colors: colors,
+        offsets: offsets,
+        tileMode: tileMode,
+        transform: this.transform,
```

Comment:
We should just premultiply this transform since it's trivial to do for linears, and set this to null here.

We should also set the unit mode to userSpaceOnUse in both return cases (but if we switch it to a bool then we should just set `objectBoundingBox` to false in both cases, since we now have coordinates that are transformed).

Reply:
done